crimsobot/cogs/image.py

```diff
@@ -33,8 +33,10 @@ async def get_image_and_embed(self, ctx: commands.Context, image: Optional[str],
                                   arg: Optional[int], title: str) -> None:
         """Get image, construct and send embed with result."""
         # process image
-        fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
-        if fp is None:
+        try:
+            fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
+        # if None is returned from utils/image.py: process_image(), then the user has already been informed
+        except TypeError:
```

Comment:
where does the TypeError come from? and if it returns None as you mention, where are we bailing out? we still need to check for None if they don't have enough balance so we're not also sending an embed with an image that doesn't exist, no?

Reply:
I am *not* sure why this has been behaving properly, but so far it is. The traceback if I remove the try/except:
```
ERROR: INVOKE ERROR
    guild: crimsoBOT's ship (552650672965943296)
     user: crimso#4547 (310618614497804289)
      msg: >acid 1 https://cdn.discordapp.com/attachments/554799675912355861/934735132571631646/IMG_5803.gif
        (crimsobot.bot)
Traceback (most recent call last):
  File "/home/anjru/.cache/pypoetry/virtualenvs/crimsobot-oMUi6eBF-py3.8/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/mnt/c/Users/handr/dev/crimsoBOT/crimsobot/cogs/image.py", line 68, in acid
    await self.get_image_and_embed(ctx, image, effect, number_of_hits, title)
  File "/mnt/c/Users/handr/dev/crimsoBOT/crimsobot/cogs/image.py", line 37, in get_image_and_embed
    fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
TypeError: cannot unpack non-iterable NoneType object```

Reply:
Oh wait, it's because `process_image()` is returning `None` instead of the expected tuple, isn't it?

Reply:
Addressed in [latest commit](https://github.com/crimsobot/crimsoBOT/pull/137/commits/c5a81dcc8961e26e9971ae6f59dfc05258ec3961).